cpp/subsys/MechanismFactory.cpp

```diff
@@ -0,0 +1,129 @@
+	}
+	return MechanismFactory::m_mechanismFactory;
+}
+
+IMechanism* MechanismFactory::GetIMechanism
+(
+	IMechanism::MECHANISM_TYPE			type		// <I> - manipulator type
+)
+{
+	IMechanism* subsys = nullptr;
+    switch ( type )
+    {
+        case IMechanism::CLIMBER:
+            subsys = m_climber;
+            break;
+
+        case IMechanism::INTAKE:
+            subsys = m_intake;
+            break;
+
+        case IMechanism::ARM:
+            subsys = m_arm;
+            break;
+
+        case IMechanism::WRIST:
+            subsys = m_wrist;
+            break;
+
+        default:
+            printf( "==>> Unknown mechanism %d \n", type );
+            break;
+    }
+
+	return subsys;
+}
+
+//=======================================================================================
+// Method:  		CreateMechanism
+// Description:		Create a mechanism from the inputs
+// Returns:         Void
+//=======================================================================================
+IMechanism* MechanismFactory::CreateMechanism
+(
+    IMechanism::MECHANISM_TYPE                   type,               // <I> - manipulator Type
+    const std::vector<IDragonMotorController*>&  motorControllers,   // <I> - Motor Controllers
+    const DragonDigitalInputVector&              digitalInputs,      // <I> - Digital Inputs
+    const DragonAnalogInputVector&               analogInputs,       // <I> - Analog Inputs
+    const DragonServoVector&                     servos              // <I> - servos
+)
+{
+    IMechanism* subsys = nullptr;
+    int index = type;
+
+    switch ( index )
+    {
+//        case IMechanism::CLIMBER:
+//            subsys = new Climber( type, motorControllers );//, digitalInputs, analogInputs );
+//            m_climber = subsys;
+//            break;
+
+        case IMechanism::INTAKE:
+            subsys = new Intake( motorControllers );
+            //subsys = new Intake( type, motorControllers, digitalInputs, analogInputs );
+            m_intake = subsys;
+            break;
+
+        case IMechanism::WRIST:
+            subsys = new Wrist( motorControllers );
+            //subsys = new Wrist( type, motorControllers, digitalInputs, analogInputs );
+            m_wrist = subsys;
+            break;
+
+        case IMechanism::ARM:
+            subsys = new Arm( motorControllers );
+            // subsys = new Arm( type, motorControllers, digitalInputs, analogInputs );
+            m_arm = subsys;
+            break;
+
+        default:
+            printf( "==>> Unknown mechanism %d \n", index );
+            break;
+    }
+
+
+    return subsys;
+}
+
+MechanismFactory::MechanismFactory() : m_climber( nullptr ),
+                                       m_intake( nullptr ),
+                                       m_arm( nullptr ),
+                                       m_wrist( nullptr )
+{
+//	m_mechanisms.resize( IMechanism::MAX_MECHANISM_TYPES );
+//	for ( auto inx=0; inx<IMechanism::MAX_MECHANISM_TYPES; ++inx )
```

Comment:
delete commented code referencing m_mechanisms

Climber is now in the master branch so you can use

IMotorControllerVector  is now available in the IMotorController so use it instead of std::vector